inc/classes/class-srm-redirect.php

```diff
@@ -106,6 +106,15 @@ public function maybe_redirect() {
 			$normalized_requested_path = $requested_path;
 		}
 
+		// normalise the request path with and without query strings, for comparison later
+		$requested_query_params = '';
+		$normalized_requested_path_no_query = false;
+		if (strpos($normalized_requested_path, "?")) {
```

Comment:
using `wp_parse_url`/`parse_url` seems to be safer

Reply:
I've updated this to use `wp_parse_url`/`parse_url`. Please take a second look.